github markdown block link fixes

shekharnwagh opened the pull request:
# Description

> [!IMPORTANT]  
> Depends on PR #198 

This PR fixes the broken internal links to other markdown docs when using "GitHub File As HTML" block.

## Changes

- Remove unused patterns for "GitHub File As HTML" block
- Allow slashes in overrides in url path when there is just one input variable. Earlier the overrides were delimited by slashes (`/`). For example in url `page_slug/$first/$second` `$first` and `$second` are assigned to first and second input variables respectively. This was a blocker in using the GitHub Doc Markdown file path as url overrides since the file path contains slashes. Now if there is just one input variable, all the url overrides are treated as just one input variable. For example `page_slug/$dir/$path/$file` will assign `$dir/$path/$file` to the input variable.
- If input variables are missing file extension then its added to allow having url overrides without the `.md` suffix
- Fix the internal links in the GitHub File As HTML block by replacing all the href attributes with local links ending in `.md`.


# Testing

- Start the local server using by following the instructions in the [Docs](https://github.com/Automattic/remote-data-blocks/blob/trunk/docs/local-development.md).
- Navigate to the page created in "GitHub File As HTML" example at the url `http://localhost:8888/gh`.
- Edit the page to add a "GitHub File As HTML" block and set the url overrides to `/gh/${file_path}`.
- Pass the file path of the markdown file in url like `http://localhost:8888/gh/${file_path}` (e.g. `http://localhost:8888/gh/docs/workflows/google-sheets-with-code`) and verify that the docs are rendered correctly and the internal links are working.

shekharnwagh committed:
remove unnecessary input variables for github html block

shekharnwagh committed:
allow slashes in url overrides for code configured pages

shekharnwagh committed:
update html links in github response

shekharnwagh committed:
lint fixes

shekharnwagh committed:
fix encoding

shekharnwagh committed:
better function names

shekharnwagh commented on example/github/remote-data-blocks/inc/queries/class-github-get-file-as-html-query.php:
@maxschmeling What do you think about extracting the query classes to shared code in `inc/Integrations/Github` directory ? GitHub would have a static response format like Shopify so this is a good candidate for this, but the current implementation is also somewhat specific to markdown.

shekharnwagh replied:
The logic to make it work without the `.md` in the path could go here - https://github.com/Automattic/remote-data-blocks/blob/bec543b65bdbb6fe68591a5a107dcdbdae451e75/example/github/remote-data-blocks/inc/queries/class-github-get-file-as-html-query.php#L70 in this file where we could append the `.md` if it is absent in the path in the input variables. We would need to make it generic though as this query is generic for GitHub and specific to just a markdown file.

maxschmeling replied:
I don't *think* we should move this to the integrations folder yet, though it might make sense to do that at some point. I think I'd want to do more with it rather than something that is narrowly scoped to what we're using it for.

shekharnwagh replied:
I've moved the markdown parsing/url update logic here in the query file and removed the GitHubParser file from integrations DIR.

shekharnwagh committed:
add optional generate field to input_schema

shekharnwagh committed:
lint fixes

shekharnwagh committed:
Merge branch 'trunk' into add/query-input-transforms

shekharnwagh committed:
feat: add default file extension support to GitHub HTML query

Enhance GitHubGetFileAsHtmlQuery with optional file extension handling. The query now accepts a default extension parameter that automatically appends to file paths when missing. Updated registration to default to '.md' for markdown files, maintaining consistency with ListFilesQuery behavior.

shekharnwagh committed:
Merge branch 'trunk' into add/query-input-transforms

shekharnwagh committed:
Merge branch 'add/query-input-transforms' into try/github-docs-links

shekharnwagh committed:
remove unused github file selector pattern

shekharnwagh committed:
make provided pattern for github file render auto-selected

shekharnwagh committed:
remove .md extension from the replaced links

shekharnwagh committed:
only update link if the file extension is md

shekharnwagh committed:
lint fixes

shekharnwagh committed:
Merge branch 'trunk' into add/query-input-transforms

shekharnwagh committed:
Merge branch 'add/query-input-transforms' into try/github-docs-links

maxschmeling reviewed (commented):
I'm wondering if this parser logic belongs in the plugin or if it should go in custom configuration in the site code. I just wonder if it's generic enough. Thoughts?

shekharnwagh committed:
refactor: rename generate to transform in query input schema

Rename 'generate' to 'transform' in query input schema and related methods to better reflect
their purpose of transforming input data. This change improves code clarity by using more
accurate terminology for data transformation operations.

shekharnwagh committed:
Merge branch 'add/query-input-transforms' into try/github-docs-links

shekharnwagh committed:
update to use fn name 'transform'

shekharnwagh committed:
move markdown parsing logic to query file

shekharnwagh committed:
Merge branch 'trunk' into add/query-input-transforms

shekharnwagh committed:
Merge branch 'add/query-input-transforms' into try/github-docs-links

shekharnwagh committed:
Merge branch 'trunk' into add/query-input-transforms

shekharnwagh committed:
Merge branch 'add/query-input-transforms' into try/github-docs-links

shekharnwagh committed:
Merge branch 'trunk' into try/github-docs-links
